diff --git src/lib.rs src/lib.rs
index 26ba271..180513e 100644
--- src/lib.rs
+++ src/lib.rs
@@ -127,7 +127,7 @@ unsafe extern "C" fn cb(msg_id: u64, client_id: u64, state: u32) {
     }
 }
 
-unsafe extern "C" fn cb_im(msg_id: u64, client_id: u64, state: u32, index_mark: *mut i8) {
+unsafe extern "C" fn cb_im(msg_id: u64, client_id: u64, state: u32, index_mark: *mut u8) {
     let index_mark = CStr::from_ptr(index_mark);
     let index_mark = index_mark.to_string_lossy().to_string();
     let state = match state {
@@ -325,7 +325,7 @@ impl Connection {
         i32_to_bool(v)
     }
 
-    pub fn wchar(&self, priority: Priority, wchar: i32) -> bool {
+    pub fn wchar(&self, priority: Priority, wchar: u32) -> bool {
         let v = unsafe { spd_wchar(self.0, priority as u32, wchar) };
         i32_to_bool(v)
     }
